modules/aggregation-layers/src/heatmap-layer/heatmap-layer.ts

```diff
@@ -540,12 +540,12 @@ export default class HeatmapLayer<
 
     const {viewport} = this.context;
 
-    triPositionBuffer.subData(packVertices(viewportCorners, 3));
+    triPositionBuffer!.subData(packVertices(viewportCorners, 3));
```

Comment:
This is being very liberal with adding exclamation marks. I agree that there are cases where the alternative would be quite verbose and it is a reasonable tradeoff.

But since they prevent the typescript compiler for detecting situations where the code would crash, I would consider wrapping in if statements in cases like this where it is easy to do. Future maintainers might thank us.

Reply:
This is the same problem as accessing `this.state.model` one every other layer. WebGL resources can be undefined in certain lifecycle methods, but are always defined in other methods.